From 866c922e1abf779a30b3818cf0873256f69c5f48 Mon Sep 17 00:00:00 2001
From: letsfindaway <me@letsfindaway.de>
Date: Fri, 20 May 2022 13:38:37 +0200
Subject: [PATCH] refactor: maintain compatibility with Qt 5.12

- make UBThumbnailWidget compatible with Qt 5.12 upwards
- use a QList for mSelectedThumbnailItems
- convert QSet to QList using values()
---
 src/gui/UBThumbnailWidget.cpp | 15 +++++++--------
 src/gui/UBThumbnailWidget.h   |  2 +-
 2 files changed, 8 insertions(+), 9 deletions(-)

diff --git a/src/gui/UBThumbnailWidget.cpp b/src/gui/UBThumbnailWidget.cpp
index 87671b781..b6b96e4a6 100644
--- a/src/gui/UBThumbnailWidget.cpp
+++ b/src/gui/UBThumbnailWidget.cpp
@@ -310,8 +310,7 @@ void UBThumbnailWidget::mousePressEvent(QMouseEvent *event)
                 int index2 = mGraphicItems.indexOf(underlyingItem);
                 if (-1 == index2)
                 {
-                    const auto items = selectedItems();
-                    mSelectedThumbnailItems = QSet<QGraphicsItem *>(items.begin(), items.end());
+                    mSelectedThumbnailItems = selectedItems();
                     return;
                 }
                 mSelectionSpan = index2 - index1;
@@ -374,8 +373,8 @@ void UBThumbnailWidget::mouseMoveEvent(QMouseEvent *event)
 
         // for horizontal moving
         const auto selectedItemsX = scene()->items(incrementXSelection, Qt::IntersectsItemBoundingRect);
-        const auto incSelectedItemsX = QSet<QGraphicsItem *>(selectedItemsX.begin(), selectedItemsX.end());
-        for (QGraphicsItem *lassoSelectedItem : incSelectedItemsX)
+
+        for (QGraphicsItem *lassoSelectedItem : selectedItemsX)
         {
             if (lassoSelectedItem)
             {
@@ -401,8 +400,8 @@ void UBThumbnailWidget::mouseMoveEvent(QMouseEvent *event)
         // for vertical moving
 
         const auto selectedItemsY = scene()->items(incrementYSelection, Qt::IntersectsItemBoundingRect);
-        const auto incSelectedItemsY = QSet<QGraphicsItem *>(selectedItemsY.begin(), selectedItemsY.end());
-        for (QGraphicsItem *lassoSelectedItem : incSelectedItemsY)
+
+        for (QGraphicsItem *lassoSelectedItem : selectedItemsY)
         {
             if (lassoSelectedItem)
             {
@@ -437,14 +436,14 @@ void UBThumbnailWidget::mouseMoveEvent(QMouseEvent *event)
             item->setSelected(false);
         }
 
-        mSelectedThumbnailItems += lassoSelectedThumbnailItems;
+        mSelectedThumbnailItems += lassoSelectedThumbnailItems.values();
         mPrevLassoRect = lassoRect;
 
         if (Qt::ControlModifier & event->modifiers())
         {
             for (int i = 0; i < mSelectedThumbnailItems.count()-1; i++)
             {
-                mSelectedThumbnailItems.values().at(i)->setSelected(true);
+                mSelectedThumbnailItems.at(i)->setSelected(true);
             }
         }
     }
diff --git a/src/gui/UBThumbnailWidget.h b/src/gui/UBThumbnailWidget.h
index ad300c120..86cdfacbc 100644
--- a/src/gui/UBThumbnailWidget.h
+++ b/src/gui/UBThumbnailWidget.h
@@ -140,7 +140,7 @@ class UBThumbnailWidget : public QGraphicsView
         int mSelectionSpan;
         QRectF mPrevLassoRect;
         QGraphicsRectItem *mLassoRectItem;
-        QSet<QGraphicsItem*> mSelectedThumbnailItems;
+        QList<QGraphicsItem*> mSelectedThumbnailItems;
         QSet<QGraphicsItem*> mPreviouslyIncrementalSelectedItemsX;
         QSet<QGraphicsItem*> mPreviouslyIncrementalSelectedItemsY;
         QElapsedTimer mClickTime;
